src/Services/ClientRegistry.php

```diff
@@ -30,20 +31,25 @@ final class ClientRegistry
     /** @var EventDispatcherInterface */
     private $eventDispatcher;
 
+    /** @var DriverOptionsInterface  */
+    private $driverOptionsService;
+
     /**
      * ClientRegistry constructor.
-     *
      * @param EventDispatcherInterface $eventDispatcher
-     * @param bool                   $debug
-     *
-     * @internal param DataCollectorLoggerInterface $logger
+     * @param DriverOptionsInterface $driverOptionsService
+     * @param bool $debug
      */
-    public function __construct(EventDispatcherInterface $eventDispatcher, bool $debug)
-    {
+    public function __construct(
+        EventDispatcherInterface $eventDispatcher,
+        bool $debug,
+        DriverOptionsInterface $driverOptionsService = null
```

Comment:
It's better `?DriverOptionsInterface $driverOptionsService` instead of `DriverOptionsInterface $driverOptionsService = null` because the parameter should be *nullable* not *optional*

Reply:
Wouldn't that create a breaking change?

Reply:
In my test I get everything work fine. What do you mean @Jean85 ? example please. I'm open to your solutions and alternatives.

Reply:
@Jean85 `ClientRegistry` is an internal service, I think it will be fine